helpers/excel.js

```diff
@@ -467,36 +467,38 @@ async function importIssues(buffer, issueContainerID, threeLeggedToken, sequenti
-                    blockedAttributeUpdates.push(key);
+                //results.failed.push({ id: issueID, row: rowNumber, error: 'Issue not found in BIM360.' });
+                tasks.push(createIssue(bim360, issueContainerID, newIssueAttributes, results));
+ 
+            }else{
+
+                const issueNumber = currentIssueAttributes.identifier;
+
+                // Check if any of the new issue properties differ from the original in BIM360, and if they *can* be changed
+                let blockedAttributeUpdates = [];
+                for (const key of Object.getOwnPropertyNames(newIssueAttributes)) {
+                    if (currentIssueAttributes[key] == newIssueAttributes[key]) { // both values are equal
+                        delete newIssueAttributes[key];
+                    } else if (!currentIssueAttributes[key] && !newIssueAttributes[key]) { // both values are "falsy" (e.g., an empty string and a null)
+                        delete newIssueAttributes[key];
+                    } else if (currentIssueAttributes.permitted_attributes.indexOf(key) === -1) { // field change not permitted
+                        blockedAttributeUpdates.push(key);
+                    }
                 }
-            }
-            // if (blockedAttributeUpdates.length > 0) {
-            //     results.failed.push({ number: issueNumber, id: issueID, error: `Changing these issue fields is not permitted: ${blockedAttributeUpdates.join(', ')}.` });
-            //     return;
-            // }
-            if (Object.getOwnPropertyNames(newIssueAttributes).length === 0) {
-                return; // No fields to update
-            }
-
-            if (sequential) {
-                tasks.push({ bim360, issueContainerID, issueID, currentIssueAttributes, newIssueAttributes, blockedAttributeUpdates, issueNumber, results });
-            } else {
-                tasks.push(updateIssue(bim360, issueContainerID, issueID, currentIssueAttributes, newIssueAttributes, blockedAttributeUpdates, issueNumber, results));
-            }
+                // if (blockedAttributeUpdates.length > 0) {
+                //     results.failed.push({ number: issueNumber, id: issueID, error: `Changing these issue fields is not permitted: ${blockedAttributeUpdates.join(', ')}.` });
+                //     return;
+                // }
+                if (Object.getOwnPropertyNames(newIssueAttributes).length === 0) {
+                    return; // No fields to update
+                }
+    
+                if (sequential) {
+                    tasks.push({ bim360, issueContainerID, issueID, currentIssueAttributes, newIssueAttributes, blockedAttributeUpdates, issueNumber, results });
+                } else {
+                    tasks.push(updateIssue(bim360, issueContainerID, issueID, currentIssueAttributes, newIssueAttributes, blockedAttributeUpdates, issueNumber, results));
```

Comment:
Is this working as expected? Here you're passing `issueID` as the 3rd parameter, but according to your updated function definition below, the third parameter should be `currentIssueAttributes`.

Reply:
oops~~ that was a mistake.  I did not want to touch previous function **updateIssues**. The new commit is submitted to restore it. Thanks.